src/parser/parser.rs

```diff
@@ -496,6 +496,18 @@ impl<'help, 'cmd> Parser<'help, 'cmd> {
                 );
             }
         }
+
+        // Conflicting arg and subcommand due to `args_conflicts_with_subcommands`.
+        let subcmd_match = self.possible_subcommand(arg_os.to_value(), &None);
+        if let Some((arg, subcmd)) = valid_arg_found.zip(subcmd_match) {
+            return ClapError::argument_subcommand_conflict(
+                self.cmd,
+                arg,
+                subcmd.into(),
+                Usage::new(self.cmd).create_usage_with_title(&[]),
+            );
+        }
```

Comment:
I was hoping to handle this before the error function, but it seems like it's not possible to error out immediately on the subcommand because it might still be a valid argument.

I'm not entirely sure if this is a bullet-proof way to catch this error, but it's the best solution I could find. So let me know if there's a better place to handle this.

Reply:
This is one of my biggest concerns with trying to improve the error in this situation. With this not being a area of priority at the moment, this means you'll need to help carry a lot of the load in analyzing and verifying this doesn't regress other parts of the user experience.

Reply:
That much should be covered both by tests and by the conditions checked before this error is emitted though, right?

Do you think there's additional changes necessary to ensure this patch does not have any unwanted side effects?

Reply:
Its also thinking of all of the different user cases and considering the error message under each one

for `todo --flag sub1` we currently give
```
error: The argument '--flag' cannot be used with subcommand 'sub1'
```

What other intentions could the user have for `todo --flag sub1`?

What other combinations of flags might hit this error but be unrelated?

Reply:
> What other combinations of flags might hit this error but be unrelated?

I mean hopefully none, that was the goal at least. I couldn't find any.

> What other intentions could the user have for todo --flag sub1?

Providing additional information on what the user might want to do is just an improvement on the error message though, right? I wouldn't say there's a good recommendation that can be made here other than removing the flag, which I'd say this error does reasonably well.

Ultimately though even if this error can be improved (maybe even in the future), I think if the question is if we can provide a good enough error here, we should also take into account that the status quo is **far** worse. So much so that it makes the option unusable (imo).